main.ts

```diff
@@ -64,4 +86,121 @@ export default class CMShowWhitespacePlugin extends Plugin {
     // @ts-ignore
     cm.setOption("showInvisibles", enable);
   }
-}
\ No newline at end of file
+
+  updateHiddenChars = () => {
+    const { showNewline, showSingleSpace, showSpace, showTab, showTrailingSpace } = this.settings;
+    const body = document.body;
+
+    if (showNewline) {
+      body.classList.remove('plugin-cm-show-whitespace-hide-newline');
+    } else {
+      body.classList.add('plugin-cm-show-whitespace-hide-newline');
+    }
```

Comment:
There's a neat function I found for `classList.toggle` which takes in a boolean value, so you could do something like this:
```suggestion
    body.classList.toggle('plugin-cm-show-whitespace-hide-newline', !showNewline);
```
I'm going to approve this PR anyway, and there's a bunch of stuff like this (not just yours) I want to tidy up, just thought I'd let you know.